### DISCUSSION TITLE
Q&A for Draft Investment 2.0.0

### DISCUSSION THREAD

**Discussion by Sumitkumar0503** [ONDC-Official/ONDC-FIS-Specifications#90] [Q&A]
Hi All,

Please feel free to raise your query & suggestion on the Investment (Mutual Fund) Draft 2.0.0.

Regards
Sumit Kumar
Team ONDC

**Comment by mohan-pace** [comment]
Hi,

I am having an issue subscribing: domain verification failed 

```
{
    "message": {
        "ack": {
            "status": "NACK"
        }
    },
    "error": {
        "type": "DOMAIN-ERROR",
        "code": "129",
        "path": null,
        "message": "https://stag-ondc.ppreciate.in : Domain verification is failed "
    }
}
```

Can someone please help: 
I am following the steps on this: https://github.com/ONDC-Official/developer-docs/blob/main/registry/Onboarding%20of%20Participants.md

Have generated the signing key pair, encryption key pair and also hosted the webpage as given:  https://stag-ondc.ppreciate.in/ondc-site-verification.html

I have verified manually that the signature created can be verified using the public key.  


```
## verification steps:
## my request id : d9b986f2-2b3c-4e39-b260-4846c7663c56
public_key64 = VerifyKey(public_key,encoder=Base64Encoder)
print("verify 1: ",public_key64.verify(signature, encoder=Base64Encoder))
## output: b'd9b986f2-2b3c-4e39-b260-4846c7663c56
```

My request to suscribe is , masking PII
```
{
    "context": {
        "operation": {
            "ops_no": 1
        }
    },
    "message": {
        "request_id": "d9b986f2-2b3c-4e39-b260-4846c7663c56",
        "timestamp": "2023-12-07T14:58:00.101Z",
        "entity": {
            "gst": {
                "legal_entity_name": <>,
                "business_address": <>,
                "city_code": [
                    "std:080"
                ],
                "gst_no": <>
            },
            "pan": {
                "name_as_per_pan": <>,
                "pan_no": <>,
                "date_of_incorporation": <>
            },
            "name_of_authorised_signatory": <>,
            "address_of_authorised_signatory": <>,
            "email_id": <>,
            "mobile_no":<>,
            "country": "IND",
            "subscriber_id": "stag-ondc.ppreciate.in",
            "unique_key_id": "d9b986f2-2b3c-4e39-b260-4846c7663c56",
            "callback_url": "/ondc/onboarding",
            "key_pair": {
                "signing_public_key": "gKtXSYyGNW4rRrWnc0ZoHyTOfKsrpAyJbRmgs6g6ldQ=",
                "encryption_public_key": "MCowBQYDK2VuAyEA2BFOJwQ0LBX++QdW0LjIAgbEXZwBalh0s9Wrho3ZKng=",
                "valid_from": "2023-12-05T00:00:00.101Z",
                "valid_until": "2024-12-04T23:59:59.101Z"
            }
        },
        "network_participant": [
            {
                "subscriber_url": "/buyerAppl",
                "domain": "nic2004:52110",
                "type": "buyerApp",
                "msn": false,
                "city_code": [
                    "std:080"
                ]
            }
        ]
    }
}
```

**Reply by 92shreyansh** [reply]
Please refer solution to this in https://github.com/ONDC-Official/developer-docs/discussions/31

In case none found please raise a comment there.

**Reply by Sumitkumar0503** [reply]
You are subscribing to an incorrect domain, it should be ONDC:FIS14 for mutual fund (12 for loan and 13 for insurance). 

Also please change the city code to "*".

**Reply by mohan-pace** [reply]
Hi,
I was on that call today, I did change the domain, but it still didn't work., they have requested to move the conversation over mail.,

**Reply by mohan-pace** [reply]
For anyone reading this , this is resolved. There we a couple of things, which i had overlooked. 

1. On the signature, we had to send this:  `signed.signature` . I was sending the signed value and not the signature. 

2. on the callback, our method has a "/" at the end, another minor misconfiguration. 
3. On the decryption, we have to return the challenge response  instead i was just logging and returning a 200 OK. 

Irrespective of the actual language on which one is developing, it will help to go through the motions and verify the following using this : https://github.com/ONDC-Official/reference-implementations/blob/main/utilities/signing_and_verification/cryptic_utils.py 

1. confirm the signed value and the verification of the signature : sign_response & verify_response methods 
2. challenge decryption : decryption method in that file ; if you want to verify a real challenge, hit the subscribe with wrong timestamp or something, you will see the challenge on the on_subscribe endpoint
